Kernel/Bus/USB/USBDevice.cpp

```diff
@@ -25,6 +26,36 @@ ErrorOr<NonnullLockRefPtr<Device>> Device::try_create(USBController const& contr
         node = move(sysfs_node);
     });
     TRY(device->enumerate_device());
+
+    // Attempt to find a driver for this device. If one is found, we call the driver's
+    // "probe" function, which initialises the local state for the device driver.
+    // It is currently the driver's responsibility to search the configuration/interface
+    // and take the appropriate action.
+    for (auto& driver : USBManagement::the().available_drivers()) {
+        if (!driver->may_accept_device(device->device_descriptor()))
+            continue;
+
+        bool has_initialized_device = false;
+        for (auto const& configuration : device->configurations()) {
+            // FIXME: We can only have one active configuration of a device at a time
+            //        We might want to have a mechanism to choose the best one available
+            for (auto const& interface : configuration.interfaces()) {
+                // FIXME: We only accept one valid interface for now,
+                //        There might be a better one available,
+                //        and interfaces may be able to operate in parallel
+                auto result = driver->probe(device, interface);
```

Comment:
Why not give each driver a ref to the list of configurations and interfaces?

Then we could skip looping over all the configs and interfaces if the device knows for certain that this is not its responsibility.

Reply:
Well, we'll have to loop either way, it really does not matter if we loop in the driver or in the Driver selection
Although technically there are presets on how many interfaces some device types should have, and some device types seem to use multiple interfaces at once, but I am not sure on how that works and even if interfaces can cooperate
We might decide later to give a driver a configuration instead of an interface, but this should be quite simple change down the line

Reply:
fair enough